### ISSUE TITLE
2ZoneDataCenterHVAC_wEconomizer East Zone does not work?

### ISSUE THREAD

**Issue by kad99kev** [NREL/EnergyPlus#10594]
Hello! Thank you for your work on EnergyPlus!

I wanted to raise an issue about controlling the setpoints of the East Zone of the Datacenter. It seems that when we change the setpoints, only the West Zone responds to the changes, the East Zone consistently has higher temperature values. I am using a Python library called Sinergym, and raise an issue there as well. But they claim it to be an EnergyPlus issue. I was wondering if you could provide some clarity behind the reason for this.

[Even when updating from 23.1 to 24.1 the issue persists](https://github.com/ugr-sail/sinergym/issues/421)
[This was the issue I raised when I initially found the issue](https://github.com/ugr-sail/sinergym/issues/395#issuecomment-1994105561)

Any advice or clarity on this would be greatly appreciated! Thank you once again!

**Comment by mjwitte** [comment]
@kad99kev Please attach a zip of idf files (and outputs if you like) so we can reproduce the problem. Also it would be helpful to mention specific results (dates/times) where you see the issue, and mention what weather file you are using.

**Comment by kad99kev** [comment]
ZIP file with IDF - [Datacenter_issue.zip](https://github.com/user-attachments/files/16159937/Datacenter_issue.zip)

I am using the USA_AZ_Davis-Monthan.AFB.722745_TMY3.epw weather file. I am running this using a reinforcement learning library called [Sinergym](https://github.com/ugr-sail/sinergym). So in RL terms, at every timestep when I send control signals to the heating and cooling setpoints, only the West Zone seems to follow the policy given, while the East Zone does not.

Thank you!

**Comment by jmarrec** [comment]
The zip you shared has a single idf, which is actually an epJSON file. It has zero output variables.

Anyways, adding a couple of output variables myself, I guess the issue you're seeing is that the Temperature of the East zone is consistently way above the thermostat cooling setpoint of 23.0, right?

![image](https://github.com/user-attachments/assets/9c44b4f3-d453-4fde-a8bf-8d7f76d6cb9e)

**Comment by jmarrec** [comment]
The same for 2ZoneDataCenterHVAC_wEconomizer.idf

![image](https://github.com/user-attachments/assets/54a86e8e-a348-4ace-810d-16e51939dca1)

**Comment by kad99kev** [comment]
Yes, that is precisely the issue that I've been noticing. I tried upgrading to EnergyPlus v24 but I noticed the same issue there as well.

**Comment by AlejandroCampoy** [comment]
Hi! Any new updates on this issue?

**Comment by dareumnam** [comment]
Thanks for reporting this.
I dug into it and found the cause. As you already know, the two zones use different air flow settings. The East zone uses a mode called _FlowControlWithApproachTemperatures_. In that mode, EnergyPlus is designed to control the supply air temperature instead of the room temperature, so it ignores the thermostat on purpose. 
So this is not actually a bug, the behavior is correct. The real problem is that nobody can find this out. The input documentation doesn't mention it, the example file doesn't mention it, and the program gives no warning.
My plan is not to change how the model works, as that would break results for other users. Instead I'm adding a warning message, plus the missing documentation and comments in the example file. I'll open a PR soon.
